src/aiida_quantumespresso/calculations/xspectra.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+"""Classes and methods for running xspectra.x with AiiDA."""
+import os
+from aiida import orm
+from aiida.common import exceptions
+from aiida.orm import Dict, FolderData, RemoteData, SinglefileData, XyData
+from aiida.common.datastructures import CalcInfo, CodeInfo
+from aiida.plugins import DataFactory
+from aiida_quantumespresso.calculations import _lowercase_dict, _uppercase_dict, _pop_parser_options
+from aiida_quantumespresso.utils.convert import convert_input_to_namelist_entry
+from aiida_quantumespresso.calculations.namelists import NamelistsCalculation
+
+KpointsData = DataFactory('array.kpoints')
+
+class XSpectraCalculation(NamelistsCalculation): 
+    """CalcJob implementation for the xspectra.x code of Quantum ESPRESSO."""
+
+    _Plotcore_FILENAME = 'plotcore.out'
+    _Spectrum_FILENAME = 'xanes.dat'
+    _default_namelists = ['INPUT_XSPECTRA', 'PLOT', 'PSEUDOS', 'CUT_OCC']
+    _blocked_keywords = [
+        ('INPUT_XSPECTRA', 'outdir', NamelistsCalculation._OUTPUT_SUBFOLDER),
+        ('INPUT_XSPECTRA', 'prefix', NamelistsCalculation._PREFIX),
+        ('PSEUDOS', 'filecore', _Plotcore_FILENAME)
+    ]
+    _internal_retrieve_list = [_Spectrum_FILENAME]
+    _retrieve_singlefile_list = []
+    _retrieve_temporary_list = []
+    _default_parser = 'quantumespresso.xspectra'
+
+    @classmethod
+    def define(cls, spec):
+        """Define the process specification."""
+
+        super().define(spec)
+
+        spec.input('parent_folder', valid_type=(orm.RemoteData, orm.FolderData), required=True)
+        spec.input('core_wfc_data', valid_type=SinglefileData,
+                   required=True,
+                   help='Core wavefunction data, generated by the upf2plotcore.sh utility')
+        spec.input('bz_sampling', valid_type=KpointsData,
```

Comment:
This was a deliberate choice from the beginning in order to clearly differentiate the k-point mesh used by PwSCF and XSpectra when running python scripts. This was done since we believed early on in the process that the convergence of the XAS spectrum w/r/t kpoints would likely be different than the total energy - and that likely we would need to establish a protocol to converge this parameter.

The original idea I had was to use a different name to avoid confusion in the inputs when this is implemented in a full workchain. However if it would be more consistent, and isn't too difficult to work with in a full workchain, then I will go with your suggestion.